src/app/options/qgscodeeditoroptions.cpp

```diff
@@ -0,0 +1,172 @@
+    {QgsCodeEditor::ColorRole::MarginBackground, mColorMarginBackground },
+    {QgsCodeEditor::ColorRole::MarginForeground, mColorMarginForeground },
+    {QgsCodeEditor::ColorRole::SelectionBackground, mColorSelectionBackground },
+    {QgsCodeEditor::ColorRole::SelectionForeground, mColorSelectionForeground },
+    {QgsCodeEditor::ColorRole::MatchedBraceBackground, mColorBraceBackground },
+    {QgsCodeEditor::ColorRole::MatchedBraceForeground, mColorBraceForeground },
+    {QgsCodeEditor::ColorRole::Edge, mColorEdge },
+    {QgsCodeEditor::ColorRole::Fold, mColorFold },
+    {QgsCodeEditor::ColorRole::Error, mColorError },
+  };
+
+  for ( auto it = mColorButtonMap.constBegin(); it != mColorButtonMap.constEnd(); ++it )
+  {
+    it.value()->setColor( QgsCodeEditor::color( it.key() ) );
+  }
+
+  mColorSchemeComboBox->addItem( tr( "Default" ), QString() );
+  mColorSchemeComboBox->addItem( tr( "Solarized (Light)" ), QStringLiteral( "solarized" ) );
+  mColorSchemeComboBox->addItem( tr( "Solarized (Dark)" ),  QStringLiteral( "solarized_dark" ) );
+  mColorSchemeComboBox->addItem( tr( "Custom" ), QStringLiteral( "custom" ) );
+
+  QgsSettings settings;
+  if ( !settings.value( QStringLiteral( "codeEditor/overrideColors" ), false, QgsSettings::Gui ).toBool() )
+  {
+    const QString theme = settings.value( QStringLiteral( "codeEditor/colorScheme" ), QString(), QgsSettings::Gui ).toString();
+    mColorSchemeComboBox->setCurrentIndex( mColorSchemeComboBox->findData( theme ) );
+  }
+  else
+  {
+    mColorSchemeComboBox->setCurrentIndex( mColorSchemeComboBox->findData( QStringLiteral( "custom" ) ) );
+  }
+
+  connect( mColorSchemeComboBox, qgis::overload<int>::of( &QComboBox::currentIndexChanged ), this, [ = ]
+  {
+    const QString theme = mColorSchemeComboBox->currentData().toString();
+    if ( theme != QStringLiteral( "custom" ) )
+    {
+      mBlockCustomColorChange = true;
+      for ( auto it = mColorButtonMap.constBegin(); it != mColorButtonMap.constEnd(); ++it )
+      {
+        it.value()->setColor( QgsCodeEditor::defaultColor( it.key(), theme ) );
+      }
+      mBlockCustomColorChange = false;
+    }
+  } );
+
+  for ( auto it = mColorButtonMap.constBegin(); it != mColorButtonMap.constEnd(); ++it )
+  {
+    connect( it.value(), &QgsColorButton::colorChanged, this, [ = ]
+    {
+      if ( mBlockCustomColorChange )
+        return;
+
+      mColorSchemeComboBox->setCurrentIndex( mColorSchemeComboBox->findData( QStringLiteral( "custom" ) ) );
+    } );
+  }
+
+  // font
+  const QFont font = QgsCodeEditor::getMonospaceFont();
+  mFontComboBox->setCurrentFont( font );
+  mSizeSpin->setValue( font.pointSize() );
+  mOverrideFontGroupBox->setChecked( !settings.value( QStringLiteral( "codeEditor/fontfamily" ), QString(), QgsSettings::Gui ).toString().isEmpty() );
+}
+
+QString QgsCodeEditorOptionsWidget::helpKey() const
+{
+  return QStringLiteral( "plugins/python_console.html" );
```

Comment:
Should we remove the help until there's a specific doc covering this generic editor settings panel?

Reply:
```suggestion
  return QStringLiteral( "introduction/qgis_configuration.html#code-editor-settings" );
```
See https://github.com/qgis/QGIS-Documentation/pull/6169 (happy to get review)